src/layout/Header/HeaderDesktop.tsx

```diff
@@ -207,8 +216,13 @@ export const HeaderDesktop = () => {
               tw="mr-[0.5em]"
               shape={ButtonShape.Pill}
               size={ButtonSize.XSmall}
-              action={ButtonAction.Primary}
+              action={
+                MustBigNumber(availableBalance?.current).gt(0)
+                  ? ButtonAction.Secondary
+                  : ButtonAction.Primary
+              }
               onClick={() => dispatch(openDialog(DialogTypes.Deposit()))}
+              state={{ isDisabled: !dydxAccounts }}
```

Comment:
thoughts on just not showing the button instead?

Reply:
ah fair - i thought it'd make the UI more jumpy, but will add to polish ideas!